tools/profiling/memory/memory_diff.py

```diff
@@ -101,19 +112,28 @@ def _run():
         text += '{}: {}\n'.format(key, value)
 else:
     print(cur, old)
-    diff_size = 0
+    call_diff_size = 0
+    channel_diff_size = 0
     for scenario in _SCENARIOS.keys():
         for key, value in sorted(_INTERESTING.items()):
             key = scenario + ': ' + key
             if key in cur:
                 if key not in old:
                     text += '{}: {}\n'.format(key, cur[key])
                 else:
-                    diff_size += cur[key] - old[key]
                     text += '{}: {} -> {}\n'.format(key, old[key], cur[key])
-
-    print("DIFF_SIZE: %f" % diff_size)
-    check_on_pr.label_increase_decrease_on_pr('per-call-memory', diff_size, 64)
+                    if key == 'call/client' or 'call/server':
+                        call_diff_size += cur[key] - old[key]
+                    else:
+                        channel_diff_size += cur[key] - old[key]
+
+    print("CALL_DIFF_SIZE: %f" % call_diff_size)
+    print("CHANNEL_DIFF_SIZE: %f" % channel_diff_size)
+    check_on_pr.label_increase_decrease_on_pr('per-call-memory', call_diff_size,
+                                              64)
+    check_on_pr.label_increase_decrease_on_pr('per-channel-memory', channel_diff_size,
+                                              1000) 
```

Comment:
Chose 1000 since the range from running it 10+ times was 17333-17758 for server, and 37012-37330 for client so max 743 difference so far. Is a range of 1000 too much?

Reply:
Check discussion thread https://github.com/grpc/grpc/pull/30507#discussion_r938940875